src/frontend/src/handler/create_source.rs

```diff
@@ -1117,10 +1116,18 @@ pub fn validate_compatibility(
     source_schema: &ConnectorSchema,
     props: &mut BTreeMap<String, String>,
 ) -> Result<()> {
-    let connector = props
+    let mut connector = props
         .get_connector()
         .ok_or_else(|| RwError::from(ProtocolError("missing field 'connector'".to_string())))?;
 
+    if connector == S3_CONNECTOR {
+        // S3 connector is deprecated, use OPENDAL_S3_CONNECTOR instead
+        // do s3 -> s3_v2 migration
+        let entry = props.get_mut(UPSTREAM_SOURCE_KEY).unwrap();
+        *entry = OPENDAL_S3_CONNECTOR.to_string();
+        connector = OPENDAL_S3_CONNECTOR.to_string();
+    }
```

Comment:
Will these lines work for an existing job?

Reply:
Only for new jobs. The existing jobs do not go through frontend again.

Reply:
xxchan's proposal https://github.com/risingwavelabs/risingwave/pull/17963#pullrequestreview-2225725091 sounds the most reasonable to me, that is, rejecting the `s3_v2` in frontend, but allow the existing ones. Can we do that?

Reply:
BTW, I think the changes here are rewriting `s3` -> `s3_v2`, which is necessary.

Reply:
> rejecting the s3_v2 in frontend, but allow the existing ones. Can we do that?

Technically, yes.
My concern is from the product side. Rejecting a higher version but accepting a lower one seems strange. And what if we evolve a new s3 source, we'd call it s3_v2 or s3_v3?

Reply:
Leaving `s3_v2` as an alias sounds acceptable to me, but please make sure all the docs & test cases are updated to `s3`.